File: 실패율.js
function solution(N, stages) {
    let rates = []  //  각 스테이지 별 실패율을 저장
    for (let i = 1; i <= N; i++) {
      let total = 0 //  해당 스테이지 보다 크거나 같은 수 개수
      let stage = 0 //  해당 스테이지의 개수
      for (let j = 0; j < stages.length; j++) {
        if (stages[j] >= i) total++
        if (stages[j] == i) stage++
      }
      rates.push([i, stage / total])  //  [스테이지, 실패율] 형태로 저장
    }
    let srates = rates.sort((a,b) => b[1]-a[1]) //  실패율 내림차순 정리
    let result = [] //  스테이지만 뽑아서 저장
    for (let i=0;i<srates.length;i++) {
      result.push(srates[i][0])
    }
    return result
}

File: 이것이 코딩 테스트다/실패율.py
N = 4
stages = [4, 4, 4, 4, 4]

rate = []
total = len(stages)
for i in range(1, N+1):
    stuck = stages.count(i)
    rate.append((i, stuck / total if total != 0 else 0))
    total -= stuck

rate.sort(key=lambda x: (-x[1], x[0]))

print(list(map(lambda x: x[0], rate)))
